fv/bpf_test.go

```diff
@@ -2270,6 +2271,48 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
 								}
 							})
 
+							if testOpts.protocol == "tcp" {
+								It("should survive conntrack cleanup sweep", func() {
+									By("checking the connectivity and thus syncing with service creation", func() {
+										cc.ExpectSome(externalClient, TargetIP(felixes[1].IP), npPort)
+										cc.CheckConnectivity()
+									})
+
+									By("monitoring a permanent connection", func() {
+										pc := &PermanentConnection{
+											Runtime:             externalClient,
+											RuntimeName:         externalClient.Name,
+											IP:                  felixes[1].IP,
+											Port:                int(npPort),
+											Protocol:            testOpts.protocol,
+											MonitorConnectivity: true,
+										}
+
+										err := pc.Start()
+										Expect(err).NotTo(HaveOccurred())
+										defer pc.Stop()
+
+										EventuallyWithOffset(1, pc.PongCount, "5s").Should(
+											BeNumerically(">", 0),
+											"Expected to see pong responses on the connection but didn't receive any")
+										log.Info("Pongs received within last 1s")
+
+										// We make sure that at least one iteration of the conntrack
+										// cleanup executes and we periodically monitor the connection if
+										// it is alive by checking that the number of PONGs keeps
+										// increasing.
+										start := time.Now()
+										prevCount := pc.PongCount()
+										for time.Since(start) < 2*proxy.ConntrackCleanerPeriod {
+											time.Sleep(time.Second)
+											newCount := pc.PongCount()
+											Expect(prevCount).Should(BeNumerically("<", newCount))
```

Comment:
Nit: if this fails, we'll get `Fail: 6 is not less than 6`, which is a bit lacking for debugging.  `Should` has an optional second arg that takes a message; recommend using that whenever you're comparing random numbers.  For example:
```
Expect(prevCount).Should(BeNumerically("<", newCount), "persistent connection stopped receiving pongs")
```